src/Host/Controllers/Catalog/ProductsController.cs

```diff
@@ -56,4 +56,13 @@ public Task<Guid> DeleteAsync(Guid id)
     {
         return Mediator.Send(new DeleteProductRequest(id));
     }
+
+    [HttpGet("export")]
+    [MustHavePermission(FSHPermissions.Products.Export)]
+    [OpenApiOperation("Export a products.", "")]
+    public async Task<FileResult> ExportAsync()
+    {
+        var result = await Mediator.Send(new ExportProductsRequest());
+        return File(result.ToArray(), "application/octet-stream", "ProductExports");
```

Comment:
Why ToArray here? Can probably return the stream immediately...

Reply:
Not possible.. Error cannot access the closed stream.